react/src/hooks/utils.tsx

```diff
@@ -20,6 +22,54 @@ export async function basicFetch(url: string, options?: FetchOptions) {
     }
 }
 
+/**
+ * Builds a data fetch request and returns a promise that resolves
+ * to the result object.
+ *
+ * @param query Query parameter provided by m-table data prop
+ * @param url The API url to request from (/api/example)
+ */
+export async function queryTableData<RowData extends object>(
+    query: MTQuery<RowData>,
+    url: string
+): Promise<MTQueryResult<RowData>> {
+    const searchParams = new URLSearchParams();
+    // add filters
+    for (let filter of query.filters) {
+        const isBoolean = ["affected", "solved"].includes("" + filter.column.field);
+        let operator = "like";
+        let value = filter.value;
+        // booleans use "eq" as operator
+        if (isBoolean) {
+            operator = "eq";
+            value = stringToBoolean(filter.value);
+        }
+        if (filter.value || isBoolean) {
+            searchParams.append("filter", `${filter.column.field};${operator};${value}`);
+        }
+    }
+    // order by
+    if (query.orderBy && query.orderDirection) {
+        searchParams.append("order_by", `${query.orderBy.field}`);
+        searchParams.append("order_dir", `${query.orderDirection}`);
+    }
+    // page information
+    searchParams.append("page", `${query.page + 1}`);
+    searchParams.append("limit", `${query.pageSize}`);
+
+    const response = await fetch(url + "?" + searchParams.toString());
+    if (response.ok) {
+        const result = await response.json();
+        return {
+            data: result.data,
+            page: result.page - 1,
```

Comment:
Okay, if we need to do `-1` and `+1` here, might it be better to change the backend to use zero-indexing? This would also make more sense with how indexing in both languages works anyway.

Reply:
I think it's fine this way. Page isn't used for indexing on the backend. It's just used to calculate the offset and needs to start at 1.

Reply:
If you look at line 56 on the backend, `offset = (int(page) * int(limit)) - int(limit)`, this is actually subtracting 1 from page as well: (page - 1) * limit

Reply:
Oh you're right. Starting at 0 makes a lot more sense then.